src/components/ToDoItem.tsx

```diff
@@ -0,0 +1,268 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable prettier/prettier */
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { DispatchContext, StateContext } from './StateContext';
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodo } from '../api/todos';
+
+type Props = {
+  todo: Todo;
+};
+
+export const ToDoItem = ({ todo }: Props) => {
+  const dispatch = useContext(DispatchContext);
+  const {
+    temporaryIds,
+    edittedTitle,
+    todos,
+  } = useContext(StateContext);
+
+  const handleDeleteTodo = (todoId: number) => {
+    dispatch({
+      type: 'LOADING_TODOS',
+      id: [todoId],
+    });
+
```

Comment:
ensure that value for focus is false before deleting 

```suggestion
    dispatch({
      type: 'FOCUS_ON_INPUT',
      value: false,
    });
```